### DISCUSSION TITLE
Many to many relation - unknown arg users

### DISCUSSION THREAD

**Discussion by david8801** [prisma/prisma#7200] [Q&A] [answered]
Hi all! Started my first project on Node + Postgre with Prisma ORM and can't manage the issue with relations.
Here's my schema with many to many relation between user and project

```prisma
model User {
  id        Int       @id @default(autoincrement())
  email     String    @unique
  password  String

  projects  Project[]
}

model Project {
  id           Int       @id @default(autoincrement())
  name         String
  color        String

  category     Category  @relation(fields: [categoryId], references: [id])
  categoryId   Int
  users        User[]
  tasks        Task[]
}
```

And here's my controller where I try to create project and assign a user to it:
```ts
class ProjectController {
    async createProject(req, res) {
        const {name, color, categoryId} = req.body;
        if (!name || !categoryId)
            return res.status(400).json({massage: 'Required fields missing'})

        try {
            let project = await prisma.project.create({
                data: {
                    name,
                    color: color ?? 'red',
                    categoryId,
                    users: {
                        connect: {id: req.user.id}
                    }
                }
            }).catch(e => res.status(500).json(e))

            res.json(project)
        } catch (e) {
            console.log(e)
            res.status(500).json(e)
        }
    }
}
```

But that's what I get after trying to create project:
![image](https://user-images.githubusercontent.com/44001477/119128561-7b0b2a00-ba3e-11eb-983a-4d757d78c0b3.png)

It says there's no user argument, but in prisma studio I can see it:
![image](https://user-images.githubusercontent.com/44001477/119128960-fd93e980-ba3e-11eb-878a-c3da0993869c.png)

Thanks in advance!

**Comment by ryands17** [comment, marked as answer]
@david8801 👋 

As you're using `connect` with the `users`, you cannot directly pass the `categoryId`. You need to use connect there as well:

```ts
class ProjectController {
    async createProject(req, res) {
        const {name, color, categoryId} = req.body;
        if (!name || !categoryId)
            return res.status(400).json({massage: 'Required fields missing'})

        try {
            let project = await prisma.project.create({
                data: {
                    name,
                    color: color ?? 'red',
                    category: { connect: { id: categoryId } },
                    users: {
                        connect: {id: req.user.id}
                    }
                }
            }).catch(e => res.status(500).json(e))

            res.json(project)
        } catch (e) {
            console.log(e)
            res.status(500).json(e)
        }
    }
}
```

**Reply by david8801** [reply]
@ryands17 Thank you very much, that helped me!

**Reply by muratx10** [reply]
@ryands17 thanks a lot. You saved me :)

**Comment by samsoft00** [comment]
This is frustrating, I spent hours trying to fix this exact issue, please help @david8801 @ryands17 

```javascript
      const ibank = await this.prisma.customerBankDetail.create({
        data: {
          customer: { connect: { id: currentUser.id } },
          account_type: { connect: { id: bankAccount.account_type_id } },
          bank_detail: {
            create: { ...bankAccount.bank_detail }
          },
          ...(bankAccount.intermediary_bank && {
            intermediary_bank: {
              create: { ...bankAccount.intermediary_bank }
            }
          })
        }
      })
```

<img width="1046" alt="Screenshot 2021-08-31 at 9 03 15 AM" src="https://user-images.githubusercontent.com/5300891/131465965-2b8d359e-6edb-42cf-a471-fc11e42cf534.png">

`clientVersion: '2.30.0'`

**Reply by ryands17** [reply]
The error is same as the one above. Don't pass foreign keys directly if you're using `create` or `connect` for other relations. Stick to the same form for all the relations.

**Reply by samsoft00** [reply]
Thank you @ryands17 
Looking at the code above, i'm not passing any foreign keys directly, i use `connect` and `create` for both relations.
I've follow example above and update my code accordingly but still encounter same error.

**Reply by ryands17** [reply]
Then you need to check if all the values you passed in `connect`/`create` are not `undefined`. This error can also show up in that case.

If that's still not the case, then share your schema here so that I can reproduce.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
